cornflow/models/meta_model.py

```diff
@@ -87,14 +87,25 @@ def patch(self, data):
         :param dict data:
         :type data:
         """
-        try:
-            self.data = jsonpatch.apply_patch(self.data, data.get("patch"))
-        except jsonpatch.JsonPatchConflict:
-            raise InvalidPatch()
-        except jsonpatch.JsonPointerException:
-            raise InvalidPatch()
+        for key, item in data.items():
+            if "patch" in key:
+                try:
+                    setattr(
+                        self,
+                        key.split("_")[0],
+                        jsonpatch.apply_patch(getattr(self, key.split("_")[0]), item),
+                    )
+                except jsonpatch.JsonPatchConflict:
+                    raise InvalidPatch()
+                except jsonpatch.JsonPointerException:
+                    raise InvalidPatch()
+            else:
+                setattr(self, key, item)
+
+        for key, item in vars(self).items():
+            if "hash" in key:
+                setattr(self, key, hash_json_256(getattr(self, key.split("_")[0])))
```

Comment:
Not sure what you want to do but this code is almost uncomprehensible. I do not like particularly:

1. the use of setattr, getattr vars(self). We need to stop using that. ([I'm not the only one who says that](https://google.github.io/styleguide/pyguide.html#219-power-features))
2. the amount of indentation is too large

Is this an attempt to do `data_patch` and `solution_patch`? Can't we just define a helper function and call it two times? once for "data_patch", another for "solution_patch" with some arguments? Maybe there's something I do not understand.

Also: are we using "patch" or "data_patch"? Because I see we changed it in the tests but here it's still "patch".

Reply:
Ok, I will think a way to make the code simplier and readable withouth using getattr setattr or vars.

It has to be name of the field_patch.